### PR TITLE
Make plugin no longer self registering

### PR DESCRIPTION
**Pull request by LeeLenaleee**
Working example: https://codepen.io/leelenaleee/pen/ExwBJgK

### CODE REVIEW

File: docs/guide/getting-started.md [hunk 1]
```diff
@@ -1,22 +1,24 @@
-# Installation
+# Getting started
```
Comment (simonbrunel) [inline docs/guide/getting-started.md:1]: nit: `Getting Started` (uppercase S)

Comment (simonbrunel) [inline docs/guide/getting-started.md:1]: The link in [guide/README.md file](https://github.com/chartjs/chartjs-plugin-deferred/blob/master/docs/guide/README.md#table-of-contents) needs to be renamed and updated. Same in the [root README.md](https://github.com/chartjs/chartjs-plugin-deferred#documentation).

File: docs/guide/getting-started.md [hunk 2]
```diff
@@ -40,3 +42,51 @@ You can download the latest version of `chartjs-plugin-deferred` from the [GitHu
...
+`chartjs-plugin-deferred` must be loaded **after** the Chart.js library!
+:::
+
+Once loaded, the plugin, available under the global `ChartDeferred` property, needs to be [registered](#registration).
+
+### Module
+
+```javascript
+import {Chart} from 'chart.js';
+import ChartDeferred from 'chartjs-plugin-datalabels';
```
Comment (simonbrunel) [inline docs/guide/getting-started.md:65]: typo: `chartjs-plugin-deferred` instead of `chartjs-plugin-datalabels`.

**Review by simonbrunel — CHANGES_REQUESTED** [review]
Awesome! Strange that you didn't have to change the plugin integration in the docs. Not sure why it was already [integrated that way](https://github.com/chartjs/chartjs-plugin-deferred/blob/master/docs/scripts/register.js#L5).

Can you also add a migration page, similar to [this one](https://github.com/chartjs/chartjs-plugin-datalabels/blob/master/docs/guide/migration.md#migration)? (mostly a copy/paste and remove parts that doesn't apply to this plugin).

**Review by simonbrunel — APPROVED** [review]
Thanks a lot @LeeLenaleee!

**Comment by gregh3269** [comment]
Hello,
I have retested this and now for me it does not load deferred.  I use the html  version, and at the moment I am using

```
new Chart(..
plugins: {
            deferred: {
              xOffset: 150,   ## defer until 150px of the canvas width are inside the viewport
              yOffset: '50%', ## defer until 50% of the canvas height are inside the viewport
              delay: 500      ## delay of 500 ms after the canvas is considered inside the viewport
            }
          }
...
```
as in the read me.  Do I have to change anything here?

The docs now mention this:
```
var chart = new Chart(ctx, {
  plugins: [ChartDeferred],
  options: {
    // ...
  }
})
```

**Comment by LeeLenaleee** [comment]
@gregh3269 do you have a reproducible sample because when I tested this it worked fine: https://codepen.io/leelenaleee/pen/ExwBJgK

**Comment by gregh3269** [comment]
Ok, I can see we now must have a

Chart.register(ChartDeferred);

first, which makes a global, guess we now cannot do it inline ?

from the getting started what does this mean/do then?

OR only to specific charts
```
var chart = new Chart(ctx, {
  plugins: [ChartDeferred],
  options: {
    // ...
  }
})
```

**Comment by LeeLenaleee** [comment]
@gregh3269 mb, seems I forgot to save the pen, updated it with inline register and it seems to work fine so if it is not working please share a sample of it: https://codepen.io/leelenaleee/pen/ExwBJgK

**Comment by gregh3269** [comment]
ok thanks, makes sense now!

```javascript
new Chart(...
  plugins: [ChartDeferred],
  ...
  options: {
    ...
    plugins: {
      deferred: {
        xOffset: 150,   ## defer until 150px of the canvas width are inside the viewport
        yOffset: '50%', ## defer until 50% of the canvas height are inside the viewport
        delay: 500      ## delay of 500 ms after the canvas is considered inside the viewport
      }
    }
  }
});
```

**Pull request merged**